src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -185,7 +184,75 @@ public override void WriteValue(object value)
         {
             ArgumentNullException.ThrowIfNull(value);
 
-            WriteString(XmlUntypedConverter.Untyped.ToString(value, _resolver));
+            Type sourceType = value.GetType();
+            switch (Type.GetTypeCode(sourceType))
+            {
+                case TypeCode.Int32:
+                    if (sourceType.IsEnum)
+                        WriteString(XmlUntypedConverter.Untyped.ToString(value, _resolver));
+                    else
+                        WriteWithBuffer((int)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.Boolean:
+                    WriteWithBuffer((bool)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.Int16:
+                    WriteWithBuffer((short)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.Int64:
+                    WriteWithBuffer((long)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.Single:
+                    WriteWithBuffer((float)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.Double:
+                    WriteWithBuffer((double)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.Decimal:
+                    WriteWithBuffer((decimal)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.DateTime:
+                    WriteWithBuffer((DateTime)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.Byte:
+                    WriteWithBuffer((byte)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.SByte:
+                    WriteWithBuffer((sbyte)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.UInt16:
+                    WriteWithBuffer((ushort)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.UInt32:
+                    WriteWithBuffer((uint)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.UInt64:
+                    WriteWithBuffer((ulong)value, XmlConvert.TryFormat);
+                    break;
+                case TypeCode.Char:
+                    WriteWithBuffer((char)value, XmlConvert.TryFormat);
+                    break;
+                default:
+                    //Guid is not supported in XmlUntypedConverter.Untyped and throws exception
```

Comment:
This limitation is really strange, but we need to adapt XmlValueConverter to allow Guid in it's methods.

Reply:
Added proposal for this [Add WriteValue(Guid/TimeSpan) in System.Xml.XmlWriter](https://github.com/dotnet/runtime/issues/75515)